src/hotspot/share/runtime/os.cpp

```diff
@@ -1037,7 +1044,17 @@ void os::print_hex_dump(outputStream* st, const_address start, const_address end
   // Print out the addresses as if we were starting from logical_start.
   while (p < end) {
     if (cols == 0) {
-      st->print(PTR_FORMAT ":   ", p2i(logical_p));
+      // highlight start of line if address of interest is located there
+      if (highlight_address != nullptr && highlight_address == p) {
```

Comment:
This only works if the highlighted address happens to fall to the start of a printed hex dump line, or? If it falls into the middle of the line, it will fail. E.g. :  `os::print_hex_dump(st, 0x10000000, 0x10000200, 8, false, 20, 0x10000008);`.

So, what you should do is to test if the highlight address falls into the range the printed line covers.

This new feature needs a gtest anyway. Cases like this should be part of gtests: highlighting the first byte, the last byte and a middle byte of the line should always highlight the line.

Reply:
Another question is how to handle this with logical addresses. Does the highlight address correspond to the logical address or the physical one? Right now, it compares the physical address, but we print the logical address.

I think if someone uses the "logical address" feature (logical_start != null), it is to hexdump something that is indepent from its physical location. E.g. the content of the CDS archive, or a network packet. In that case the caller should specify a logical address as highlight address, I think that makes the most sense.